ScoutSuite/providers/utils.py

```diff
@@ -45,12 +46,15 @@ def run_concurrently(func):
     if len(entities) == 0:
         return
 
-    tasks = {
-        asyncio.ensure_future(
-            get_and_set_func(entity, **kwargs)
-        ) for entity in entities for get_and_set_func in get_and_set_funcs
-    }
-    await asyncio.wait(tasks)
+    try:
+        tasks = {
+            asyncio.ensure_future(
+                get_and_set_func(entity, **kwargs)
+            ) for entity in entities for get_and_set_func in get_and_set_funcs
+        }
+        await asyncio.wait(tasks)
+    except Exception as e:
```

Comment:
I'm afraid that with this `try... except` clause, any exception that could occured during the execution of `func` will be catched and thus, won't be propagated any further.
It would mean that all of the exception handling done above for AWS facades will never be used and that in case of failures during data fetching, the only kind of logs you would get would be `Failed to run function concurrently: ...`. 

Or did you test it and I'm completely wrong?